### ISSUE TITLE
Cannot use JsonType with GraalVM

### ISSUE THREAD

**Issue by bastiansmn** [vladmihalcea/hypersistence-utils#748]
**Describe the bug**
In my Java Entity, I added a field of type Map<String, String>. I want it to be a jsonb type in my PSQL database, so I added the `@Type(JsonBinaryType.class)` annotation. It works great locally, builded with GraalVM JDK running in the JVM, but as soon as I deploy my app to a Dockerized native image, it doesn't start with the following error :

```
Could not instantiate Type: io.hypersistence.utils.hibernate.type.json.Json 
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1770) ~[app:6.0.13]                             
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:598) ~[app:6.0.13]                                
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:520) ~[app:6.0.13]                                  
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:325) ~[app:6.0.13]                                                        
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[app:6.0.13]                                            
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:323) ~[app:6.0.13]                                                                 
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199) ~[app:6.0.13]                                                                   
        at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1166) ~[app:6.0.13]                                                          
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:940) ~[app:6.0.13]                                   
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:616) ~[app:6.0.13]                                                           
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[app:3.0.12]                                  
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:733) ~[app:3.0.12]                                                                                        
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:435) ~[app:3.0.12]                                                                                 
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:311) ~[app:3.0.12]                                                                                            
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1301) ~[app:3.0.12]                                                                                           
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1290) ~[app:3.0.12]                                                                                           
        at fr.bastiansmn.laponiegouv.LaponieGouvApplication.main(LaponieGouvApplication.java:10) ~[app:na]                                                                                     
        at java.base@21.0.2/java.lang.invoke.LambdaForm$DMH/sa346b79c.invokeStaticInit(LambdaForm$DMH) ~[na:na]                                                                                
         Caused by: org.hibernate.MappingException: Could not instantiate Type: io.hypersistence.utils.hibernate.type.json.JsonType                                                                 
        at org.hibernate.boot.model.TypeBeanInstanceProducer.produceBeanInstance(TypeBeanInstanceProducer.java:47) ~[na:na]                                                                    
        at org.hibernate.boot.model.TypeBeanInstanceProducer.produceBeanInstance(TypeBeanInstanceProducer.java:53) ~[na:na]                                                                    
        at org.springframework.orm.hibernate5.SpringBeanContainer.createBean(SpringBeanContainer.java:187) ~[na:na]                                                                            
        at org.springframework.orm.hibernate5.SpringBeanContainer.getBean(SpringBeanContainer.java:125) ~[na:na]                                                                               
        at org.hibernate.resource.beans.internal.ManagedBeanRegistryImpl.getBean(ManagedBeanRegistryImpl.java:110) ~[app:6.1.7.Final]                                                          
        at org.hibernate.mapping.BasicValue.setExplicitCustomType(BasicValue.java:825) ~[app:6.1.7.Final]                                                                                      
        at org.hibernate.cfg.annotations.BasicValueBinder.fillSimpleValue(BasicValueBinder.java:1238) ~[na:na]                                                                                 
        at org.hibernate.cfg.SetBasicValueTypeSecondPass.doSecondPass(SetBasicValueTypeSecondPass.java:26) ~[na:na]                                                                            
        at org.hibernate.boot.internal.InFlightMetadataCollectorImpl.processSecondPasses(InFlightMetadataCollectorImpl.java:1782) ~[na:na]                                                     
        at org.hibernate.boot.internal.InFlightMetadataCollectorImpl.processSecondPasses(InFlightMetadataCollectorImpl.java:1729) ~[na:na]                                                     
        at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.complete(MetadataBuildingProcess.java:300) ~[na:na]
at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.metadata(EntityManagerFactoryBuilderImpl.java:1348) ~[app:6.1.7.Final]                                              
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:1419) ~[app:6.1.7.Final]                                                 
        at org.springframework.orm.jpa.vendor.SpringHibernateJpaPersistenceProvider.createContainerEntityManagerFactory(SpringHibernateJpaPersistenceProvider.java:75) ~[na:na]                
        at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.createNativeEntityManagerFactory(LocalContainerEntityManagerFactoryBean.java:376) ~[app:6.0.13]                  
        at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.buildNativeEntityManagerFactory(AbstractEntityManagerFactoryBean.java:409) ~[app:6.0.13]                               
        at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.afterPropertiesSet(AbstractEntityManagerFactoryBean.java:396) ~[app:6.0.13]                                            
        at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.afterPropertiesSet(LocalContainerEntityManagerFactoryBean.java:352) ~[app:6.0.13]                                
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1817) ~[app:6.0.13]                          
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1766) ~[app:6.0.13]                             
        ... 17 common frames omitted                                                                                                                                                           
         Caused by: java.lang.InstantiationException: io.hypersistence.utils.hibernate.type.json.JsonType                                                                                           
        at java.base@21.0.2/java.lang.Class.newInstance(DynamicHub.java:719) ~[app:na]                                                                                                         
        at org.hibernate.boot.model.TypeBeanInstanceProducer.produceBeanInstance(TypeBeanInstanceProducer.java:42) ~[na:na]                                                                    
        ... 36 common frames omitted                                                                                                                                                           
         Caused by: java.lang.NoSuchMethodException: io.hypersistence.utils.hibernate.type.json.JsonType.<init>()                                                                                   
        at java.base@21.0.2/java.lang.Class.checkMethod(DynamicHub.java:1075) ~[app:na]                                                                                                        
        at java.base@21.0.2/java.lang.Class.getConstructor0(DynamicHub.java:1238) ~[app:na]                                                                                                    
        at java.base@21.0.2/java.lang.Class.newInstance(DynamicHub.java:706) ~[app:na]                                                                                                         
        ... 37 common frames omitted
```

**To Reproduce**

Hibernate versions : 
 hibernate-commons-annotations 6.0.6.Final
 hibernate-core 6.1.7.Final
 hypersistence-utils-hibernate-60 3.8.3

Add this field to an entity : 
```
@Type(JsonBinaryType.class)
    @Column(columnDefinition = "jsonb")
    private Map<String, String> draw;
```

**Expected behavior**
Working with native image as it does with JVM.

**Additional context**

**Comment by vladmihalcea** [comment]
There is no support for GraalVM or native images. You will have to investigate and see what has to be changed in order to make it work.

**Comment by vladmihalcea** [comment]
This might have been fixed by #740
